### PR TITLE
Remove dependency to six

### PR DESCRIPTION
**Pull request by niboshi**

### CODE REVIEW

File: cupy/cuda/memory.pyx [hunk 1]
```diff
@@ -657,17 +657,15 @@ cdef class PooledMemory(BaseMemory):
                 device_id = self.device_id
                 pmem_id = id(self)
 
-                # avoid six for performance
-                hooks_values = hooks.values()
-                for hook in hooks_values:
+                for hook in hooks.values:
```
Comment (toslunar) [inline cupy/cuda/memory.pyx]:
```suggestion
                for hook in hooks.values():
```

Comment (niboshi) [reply cupy/cuda/memory.pyx]: Thanks. Force-pushed.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by leofang** [comment]
This is great! Definitely the right thing to do to reduce code complexity now that we're on PY3 only. Thanks @niboshi.

**Pull request merged**